compiler/src/dotty/tools/dotc/transform/init/Errors.scala

```diff
@@ -89,20 +82,16 @@ object Errors {
   }
 
   /** Promote a value under initialization to fully-initialized */
-  case class UnsafePromotion(msg: String, source: Tree, trace: Seq[Tree], errors: Errors) extends Error {
-    assert(errors.nonEmpty)
+  case class UnsafePromotion(msg: String, source: Tree, trace: Seq[Tree], error: Error) extends Error {
```

Comment:
I see the call site we have many errors -- so maybe keep the original `errors: Errors` is OK, but only show the first one.